src/tools/imgtool/imgtool.cpp

```diff
@@ -1125,81 +1110,78 @@ std::string imgtool::image::info()
 //  into a NUL delimited list
 //-------------------------------------------------
 
-imgtoolerr_t imgtool::partition::cannonicalize_path(uint32_t flags, const char **path, char **alloc_path)
+imgtoolerr_t imgtool::partition::cannonicalize_path(uint32_t flags, const char *path, std::string &result)
 {
 	imgtoolerr_t err = (imgtoolerr_t)IMGTOOLERR_SUCCESS;
-	char *new_path = nullptr;
-	char path_separator, alt_path_separator;
-	const char *s;
-	int in_path_separator, i, j;
 
-	path_separator = m_path_separator;
-	alt_path_separator = m_alternate_path_separator;
+	// is this path NULL?  if so, is that ignored?
+	if (!path && (flags & PATH_LEAVENULLALONE))
+		return IMGTOOLERR_SUCCESS;
 
-	/* is this path NULL?  if so, is that ignored? */
-	if (!*path && (flags & PATH_LEAVENULLALONE))
-		goto done;
-
-	/* is this the special filename for bootblocks? */
-	if (*path == FILENAME_BOOTBLOCK)
+	// is this the special filename for bootblocks?
+	if (path == FILENAME_BOOTBLOCK)
 	{
 		if (!(flags & PATH_CANBEBOOTBLOCK))
 			err = (imgtoolerr_t)IMGTOOLERR_UNEXPECTED;
 		else if (!m_supports_bootblock)
 			err = (imgtoolerr_t)IMGTOOLERR_FILENOTFOUND;
-		goto done;
+		return err;
+	}
+
+	// normalize the path into the native character set
+	std::string converted_path;
+	imgtool::charconverter *charconverter = (imgtool::charconverter *) get_info_ptr(IMGTOOLINFO_PTR_CHARCONVERTER);
```

Comment:
Can you please change this to an explicit `reinterpret_cast` rather than a C-style cast?  It makes it easier to grep for various kinds of pointer (ab)use.

Also, having `imgtool::charconverter` and local `charconverter` in the same line is a bit of a red flag.  Want to change the name of the local to `converter` or something to make it a bit clearer?

And you might want to make it `imgtool::charconverter *const` since you're only making one attempt to find one, so the thing shouldn't be changed afterwards.